### PR TITLE
[dotnet] Make it clearer that the linker is off by default for Mac Catalyst (like it already is for macOS).

### PR DESCRIPTION
**Pull request by rolfbjarne**
There's no real change here, the code did exactly this in before, but it was a
side effect of '_SdkIsSimulator' being 'true' for Mac Catalyst, which isn't
obvious. Now the logic is explicit for all platforms.

### CODE REVIEW

File: dotnet/targets/Xamarin.Shared.Sdk.targets [hunk 1]
```diff
@@ -872,9 +872,9 @@
...
 		<PropertyGroup>
-			<_DefaultLinkMode Condition="'$(_PlatformName)' == 'macOS'">None</_DefaultLinkMode> <!-- Linking is off by default for macOS apps -->
-			<_DefaultLinkMode Condition="'$(_PlatformName)' != 'macOS' And '$(_SdkIsSimulator)' == 'true'">None</_DefaultLinkMode> <!-- Linking is off by default in the simulator -->
-			<_DefaultLinkMode Condition="'$(_PlatformName)' != 'macOS' And '$(_SdkIsSimulator)' != 'true'">SdkOnly</_DefaultLinkMode> <!-- Linking is SdkOnly for iOS/tvOS/watchOS apps on device -->
+			<_DefaultLinkMode Condition="'$(_PlatformName)' == 'macOS' Or '$(_PlatformName)' == 'MacCatalyst'">None</_DefaultLinkMode> <!-- Linking is off by default for macOS and Mac Catalyst apps -->
+			<_DefaultLinkMode Condition="'$(_PlatformName)' != 'macOS' And '$(_PlatformName)' == 'MacCatalyst' And '$(_SdkIsSimulator)' == 'true'">None</_DefaultLinkMode> <!-- Linking is off by default in the simulator -->
+			<_DefaultLinkMode Condition="'$(_PlatformName)' != 'macOS' And '$(_PlatformName)' == 'MacCatalyst' Abd '$(_SdkIsSimulator)' != 'true'">SdkOnly</_DefaultLinkMode> <!-- Linking is SdkOnly for iOS/tvOS/watchOS apps on device -->
```
Comment (spouliot) [inline dotnet/targets/Xamarin.Shared.Sdk.targets:877]: `Adb` won't work :)

Comment (spouliot) [reply dotnet/targets/Xamarin.Shared.Sdk.targets:877]: should not it be `'$(_PlatformName)' != 'MacCatalyst'`
since it's for everything but Mac* ?

**Review by spouliot — CHANGES_REQUESTED** [review]

**Comment by spouliot** [comment]
I'm not sure if that not-linking default is still the right choice considering that AOT is required (at least on ARM targets) and the build will become very long...

App size will also be quite larger, but that's likely not an issue (and it can be changed to linking if/when it becomes one)

**Comment by rolfbjarne** [comment]
> I'm not sure if that not-linking default is still the right choice considering that AOT is required (at least on ARM targets) and the build will become very long...

The interpreter still works, and that's the default (it still requires AOT-compiling System.Private.CoreLib.dll, but that's it).

**Comment by spouliot** [comment]
> > I'm not sure if that not-linking default is still the right choice considering that AOT is required (at least on ARM targets) and the build will become very long...
> 
> The interpreter still works, and that's the default (it still requires AOT-compiling System.Private.CoreLib.dll, but that's it).

That (interpreter) make sense for debug builds but I was thinking about release builds

**Review by spouliot — APPROVED** [review]

**Comment by rolfbjarne** [comment]
Test failure is unrelated (https://github.com/xamarin/maccore/issues/2451).

**Comment by rolfbjarne** [comment]
> > > I'm not sure if that not-linking default is still the right choice considering that AOT is required (at least on ARM targets) and the build will become very long...
> > 
> > 
> > The interpreter still works, and that's the default (it still requires AOT-compiling System.Private.CoreLib.dll, but that's it).
> 
> That (interpreter) make sense for debug builds but I was thinking about release builds

Yes, I'm not sure what's the best default behavior here. I filed #12264 so that we can make a decision.

**Pull request merged**